tests/soldier.test.ts

```diff
@@ -0,0 +1,115 @@
+import request from 'supertest';
+import dotenv from "dotenv";
+import { Express } from 'express-serve-static-core';
+import { Soldier as MongoSoldier} from "../repositories/schemas/soldier.schema";
+import { getServer, startMongoConnection } from "../app";
+
+dotenv.config();
+
+let server: Express;
+
+beforeAll(async () => {
+  startMongoConnection();
+  server = getServer();
+})
+
+beforeEach(async () => {
+  await MongoSoldier.deleteMany({});
+})
+
+afterAll(async () => {
+  MongoSoldier.deleteMany({});
+})
+
+describe("POST /soldiers", () => {
+
+  test("returns code 201 and object if all info passed correctly", async () => {
+    const res = await request(server).post(`/soldiers`).send({
+      _id: "1111",
+      name: "ilan",
+      rank: "rabat",
+      limitations: ["limit1", "limit2"]
+    })
+    expect(res.statusCode).toEqual(201);
+    expect(res.body).toEqual([{
+        _id: "1111",
+        name: "ilan",
+        rank: "rabat",
+        limitations: ["limit1", "limit2"],
+        duties: []
+      }]);
+  });
+
+  test("returns code 400 if missing a field", async () => {
+      const res = await request(server).post(`/soldiers`).send({
+        _id: "1111",
+        rank: "rabat",
+        limitations: ["limit1", "limit2"]
+      });
+      expect(res.statusCode).toEqual(400);
+  });
+
+});
+
+describe("GET /soldiers", () => {
+
+  const toInsert = [
```

Comment:
If you want, you can create a function that generates soldiers for tests, but not mandatory.

Consider putting this at the top and using it everywhere